StreamReader.h

```diff
@@ -2,28 +2,35 @@
 
 #include <fstream>
 #include <string>
+#include <cstdint>
 
 class StreamReader {
 public:
 	virtual ~StreamReader() = 0;
-	virtual void Read(char* buffer, size_t size) = 0;
+	virtual void Read(char* buffer, uint64_t size) = 0;
```

Comment:
I think this should remain `size_t`, as memory reads are always limited by addressable memory size.